crates/config/src/sections/upstream_oauth2.rs

```diff
@@ -472,6 +482,17 @@ pub struct Provider {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub token_endpoint_auth_signing_alg: Option<JsonWebSignatureAlg>,
 
+    /// Expected signature for the JWT payload returned by the token
+    /// authentication endpoint.
+    ///
+    /// If null, the response is expected to be an unsigned JSON payload.
+    /// Defaults to `RS256`.
+    #[serde(
+        default = "signed_response_alg_default",
+        skip_serializing_if = "is_signed_response_alg_default"
+    )]
+    pub id_token_signed_response_alg: Option<JsonWebSignatureAlg>,
```

Comment:
`None` is probably invalid for the `id_token`? I don't think this should be optional